src/scripts/fetchOrbitChainsData.ts

```diff
@@ -0,0 +1,123 @@
+import * as fs from 'fs';
+import { L2Network, constants as arbConstants } from '@arbitrum/sdk';
+
+type OrbitChainData = {
+  chainId: number;
+  confirmPeriodBlocks: number;
+  ethBridge: {
+    bridge: string;
+    inbox: string;
+    outbox: string;
+    rollup: string;
+    sequencerInbox: string;
+  };
+  nativeToken: string;
+  explorerUrl: string;
+  rpcUrl: string;
+  isCustom: boolean;
+  isTestnet: boolean;
+  name: string;
+  slug: string;
+  parentChainId: number;
+  tokenBridge: {
+    parentCustomGateway: string;
+    parentErc20Gateway: string;
+    parentGatewayRouter: string;
+    parentMultiCall: string;
+    parentProxyAdmin: string;
+    parentWeth: string;
+    parentWethGateway: string;
+    childCustomGateway: string;
+    childErc20Gateway: string;
+    childGatewayRouter: string;
+    childMultiCall: string;
+    childProxyAdmin: string;
+    childWeth: string;
+    childWethGateway: string;
+  };
+  bridgeUiConfig: {
+    color: string;
+    network: {
+      name: string;
+      logo: string;
+      description: string;
+    };
+    nativeTokenData: {
+      name: string;
+      symbol: string;
+      logoUrl: string;
+    };
+  };
+};
+
+type OrbitChainDataResponse = {
+  mainnet: OrbitChainData[];
+  testnet: OrbitChainData[];
+};
+
+type L2NetworkWithRpc = L2Network & { rpcUrl: string };
+
+const fileName = './src/Assets/orbitChainsData.json';
+
+export async function fetchOrbitChainsData() {
+  const response = await fetch(
+    'https://raw.githubusercontent.com/OffchainLabs/arbitrum-token-bridge/refs/heads/master/packages/arb-token-bridge-ui/src/util/orbitChainsData.json',
+  );
+
+  const data: OrbitChainDataResponse = await response.json();
+  return data.mainnet.concat(data.testnet);
+}
+
+function parseChainToL2Network({
+  chainId,
+  nativeToken,
+  isTestnet,
+  slug,
+  bridgeUiConfig,
+  parentChainId,
+  tokenBridge,
+  ...chain
```

Comment:
hm in this situation I'd maybe not do spread rest, to make it clearer what data is being mapped